Mathlib/RingTheory/Nilpotent.lean

```diff
@@ -55,6 +55,18 @@ theorem IsNilpotent.neg [Ring R] (h : IsNilpotent x) : IsNilpotent (-x) := by
   rw [neg_pow, hn, mul_zero]
 #align is_nilpotent.neg IsNilpotent.neg
 
+lemma IsNilpotent.pow {n : ℕ} {S : Type*} [MonoidWithZero S] {x : S}
+    (hx : IsNilpotent x) : IsNilpotent (x ^ n.succ) := by
+  obtain ⟨N,hN⟩ := hx
```

Comment:
```suggestion
  obtain ⟨N, hN⟩ := hx
```

Reply:
Many thanks for your comments.

Reply:
Thanks for pointing out that this is true works for SemigroupWithZero. Unfortunately there doesn't seem to be an instance of Pow S Nat for S a SemigroupWithZero, and I suspect that I'd break things if I defined one. I think I'll leave it as it is.